crates/task-impls/src/network.rs

```diff
@@ -160,11 +163,14 @@ impl<TYPES: NodeType> NetworkMessageTaskState<TYPES> {
 
             // Handle external messages
             MessageKind::External(data) => {
+                if sender == self.public_key {
```

Comment:
Does this essentially prevent loops with processing messages that we sent?

Reply:
It's basically so we don't broadcast messages to ourselves. I feel like it's unexpected behavior from the applications perspective to get it's own message